packages/sails-ng-common/src/config/component/checkbox-input.model.ts

```diff
@@ -0,0 +1,27 @@
+import {BaseFormFieldModelConfig, BaseFormFieldModelDefinition} from "../form-field-model.model";
+import {BaseFormFieldComponentConfig, BaseFormFieldComponentDefinition} from "../form-field-component.model";
+
+
+export type CheckboxModelValueType = string | boolean | null | Array<any>;
+
+export interface CheckboxInputComponentDefinition extends BaseFormFieldComponentDefinition {
+    class: "CheckboxInputComponent";
+    config?: CheckboxInputComponentConfig;
+}
+
+export class CheckboxInputComponentConfig extends BaseFormFieldComponentConfig {
+    public placeholder?: string = '';
+    public options: Array<{ label: string; value: any; disabled?: boolean }> = [];
```

Comment:
The array type here should be CheckboxOption, after that interface is moved into `sails-ng-common`. Same for the other component config options types.